command/sub8_missions/missions/roulette_wheel.py

```diff
@@ -0,0 +1,204 @@
+            for target in self.targets. Eventually we will want to take the
+            best target, which is the TCX target. Priority targetting will
+            come once we confirm we can actually unblock currently blocked
+            targets.
+            '''
+            res = yield self.sub.vision_proxies.arm_torpedos.get_pose(
+                target='roulette_wheel')
+            if res.found:
+                self.ltime = res.pose.header.stamp
+                self.targets[target].update_position(
+                    rosmsg_to_numpy(res.pose.pose.position))
+                marker = Marker(
+                    ns='roulette_wheel',
+                    action=visualization_msgs.Marker.ADD,
+                    type=Marker.ARROW,
+                    scale=Vector3(0.2, 0.5, 0),
+                    points=np.array([Point(0, 0, 0),
+                                     res.pose.pose.position]))
+                marker.id = 3
+                marker.header.frame_id = '/base_link'
+                marker.color.r = 1
+                marker.color.g = 0
+                marker.color.a = 1
+                markers.markers.append(marker)
+            if self.targets[target].position is not None:
+                info += target + ' '
+            yield self.sub.nh.sleep(0.5)  # Throttle service calls
+            self.print_info(info)
+
+    @util.cancellableInlineCallbacks
+    def pattern(self):
+        def err():
+            self.print_info('Search pattern canceled')
+
+        self.pattern_done = False
+        for i, move in enumerate(self.moves[self.move_index:]):
+            move = self.sub.move.relative(np.array(move)).go(blind=self.BLIND, speed=0.1)
+            move.addErrback(err)
+            yield move
+            self.move_index = i + 1
+        self.print_bad('Pattern finished. Firing at any locked targets.')
+        self.pattern_done = True
+
+    @util.cancellableInlineCallbacks
+    def fire(self, target):
+        self.print_info("DROPPING {}".format(target))
+        target_pose = self.targets[target].position
+        yield self.sub.move.go(blind=self.BLIND, speed=0.1)  # Station hold
+        transform = yield self.sub._tf_listener.get_transform('/map', '/base_link')
+        target_position = transform._q_mat.dot(
+                target_pose - transform._p)
+
+        sub_pos = yield self.sub.tx_pose()
+        # print('Current Sub Position: ', sub_pos)
+
+        # sub_pos = transform._q_mat.dot(
+                # (sub_pos[0]) - transform._p)
+        # target_position = target_position - sub_pos[0]
+        # yield self.sub.move.look_at_without_pitching(target_position).go(
+            # blind=self.BLIND, speed=.1)
+        # print('Map Position: ', target_position)
+        yield self.sub.move.relative(np.array([0, target_position[1], 0])).go(blind=True, speed=.1)
+        yield self.sub.move.relative(np.array([target_position[0], 0, target_position[2]+.5])).go(
```

Comment:
TBH, it'll probably be better to hardcode a defined depth

Reply:
`sub.move.depth(x)`

Looks like it'll be 0.6m above the floor according to the rules